networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_cfg_syncer.py

```diff
@@ -0,0 +1,207 @@
+                dev_owner = intf['device_owner']
+                dev_id = intf['device_id'][0:6]
+                ip_addr = intf['fixed_ips'][0]['ip_address']
+                intf_info = "    INTF: %s, %s, %s" % \
+                            (ip_addr, dev_id, dev_owner)
+                LOG.info(intf_info)
+
+        running_cfg = self.get_running_config(conn)
+        parsed_cfg = HTParser(running_cfg)
+
+        invalid_cfg = []
+
+        invalid_cfg += self.clean_snat(conn,
+                                       router_id_dict,
+                                       intf_segment_dict,
+                                       segment_nat_dict,
+                                       parsed_cfg)
+
+        invalid_cfg += self.clean_nat_pool_overload(conn,
+                                                    router_id_dict,
+                                                    intf_segment_dict,
+                                                    segment_nat_dict,
+                                                    parsed_cfg)
+
+        invalid_cfg += self.clean_nat_pool(conn,
+                                           router_id_dict,
+                                           intf_segment_dict,
+                                           segment_nat_dict,
+                                           parsed_cfg)
+
+        if (cfg.CONF.multi_region.enable_multi_region):
+            set_route_regex = SET_ROUTE_MULTI_REGION_REGEX
+        else:
+            set_route_regex = SET_ROUTE_REGEX
+
+        invalid_cfg += self.clean_routes(conn,
+                                       router_id_dict,
+                                       intf_segment_dict,
+                                       segment_nat_dict,
+                                       parsed_cfg,
+                                       set_route_regex)
+        invalid_cfg += self.clean_acls(conn,
+                                       intf_segment_dict,
+                                       segment_nat_dict,
+                                       parsed_cfg)
+
+        invalid_cfg += self.clean_interfaces(conn,
+                                             intf_segment_dict,
+                                             segment_nat_dict,
+                                             parsed_cfg)
+
+        invalid_cfg += self.clean_vrfs(conn, router_id_dict, parsed_cfg)
+        LOG.debug("invalid_cfg = %s " % (pp.pformat(invalid_cfg)))
+        return invalid_cfg
+
+    def clean_routes(self,
+                     conn,
+                     router_id_dict,
+                     intf_segment_dict,
+                     segment_nat_dict,
+                     parsed_cfg,
+                     route_regex):
+        delete_route_list = []
+        routes = parsed_cfg.find_objects(route_regex)
+        for route in routes:
+            LOG.info(_LI("\ndefault route: %s"), (route))
+            match_obj = re.match(route_regex, route.text)
```

Comment:
why we have to apply the regex again here? earlier at line#133 we have done this: routes = parsed_cfg.find_objects(route_regex)
to get the routes....